src/modm/driver/inertial/ixm42xxx_data.hpp

```diff
@@ -0,0 +1,251 @@
+// coding: utf-8
+// ----------------------------------------------------------------------------
+/*
+ * Copyright (c) 2023, Rasmus Kleist Hørlyck Sørensen
+ *
+ * This file is part of the modm project.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+// ----------------------------------------------------------------------------
+
+#ifndef MODM_IXM42XXX_DATA_HPP
+#define MODM_IXM42XXX_DATA_HPP
+
+#include <cstring>
+#include <span>
+
+#include <modm/math.hpp>
+#include <modm/math/utils/endianness.hpp>
+
+namespace modm
+{
+
+template < class Transport >
+class Ixm42xxx;
+
+namespace ixm42xxxdata
+{
+
+/// @ingroup modm_driver_ixm42xxx
+struct
+Data
+{
+    template < class Transport >
+    friend class ::modm::Ixm42xxx;
+
+    Data(std::span<uint8_t> fifoBuffer = std::span<uint8_t>()) :
+        fifoCount(0),
+        fifoBuffer(fifoBuffer),
+        accelScale(16.f),
+        gyroScale(2000.f)
+    {
+    }
+
+    // DATA ACCESS
+    ///@{
+    int16_t inline
+    getTemp()
+    { return sensorData.temp; }
+
+    void inline
+    getTemp(int16_t &temp)
+    { temp = getTemp(); }
+
+    void inline
+    getTemp(float &temp)
+    { temp = getTemp() / 132.48f + 25.f; }
+
+    Vector3i inline
+    getAccel()
+    { return Vector3i(sensorData.accel); }
+
+    void inline
+    getAccel(Vector3i &accel)
+    { accel = getAccel(); }
+
+    void inline
+    getAccel(Vector3f &accel)
+    { accel = getAccel() * accelScale / INT16_MAX; }
+
+    Vector3i inline
+    getGyro()
+    { return Vector3i(sensorData.gyro); }
+
+    void inline
+    getGyro(Vector3i &gyro)
+    { gyro = getGyro(); }
+
+    void inline
+    getGyro(Vector3f &gyro)
+    { gyro = getGyro() * gyroScale / INT16_MAX; }
+    ///@}
+
+    constexpr uint16_t
+    getFifoCount() const
+    { return fifoCount; }
+
+    constexpr uint16_t
+    getFifoBufferSize() const
+    { return fifoBuffer.size(); }
+
+    constexpr std::span<const uint8_t>
+    getFifoData() const
+    { return fifoBuffer.subspan(0, fifoCount); }
+
+private:
+    struct
+    SensorData {
+        int16_t temp;
+        int16_t accel[3];
+        int16_t gyro[3];
+    } sensorData;
+
+    uint16_t fifoCount;
+    std::span<uint8_t> fifoBuffer;
+
+    float accelScale;
+    float gyroScale;
+};
+
+/// @ingroup modm_driver_ixm42xxx
+struct
+FifoPacket
+{
+
+    FifoPacket() : header(0), accel(0), gyro(0), temp(0), timestamp(0), extension(0) {}
```

Comment:
```suggestion
    constexpr FifoPacket() : header(0), accel(0), gyro(0), temp(0), timestamp(0), extension(0) {}
```
You can probably mark all functions constexpr.